src/tennissimulator/menu/Menu.java

```diff
@@ -100,61 +109,58 @@ private Gender resolveGender(String gender) {
 
     private int promptUserForRating(Organization organization) {
         System.out.print("Enter a " + organization.toString() + " rating: ");
+
         return scanner.nextInt();
     }
 
     private String promptUserForName() {
         System.out.print("Enter a full player name: ");
+
         return scanner.nextLine();
     }
 
     private String promptUserForGender() {
         System.out.print("Enter a gender (man/woman): ");
+
+        return scanner.nextLine();
+    }
+
+    private String promptUserForId() {
+        System.out.println("\nChoose player ID: ");
+
         return scanner.nextLine();
     }
 
     private void selectPlayersForMatchSimulation() {
-        playersDatabase.print();
         if (playersDatabase.getPlayers().isEmpty()) {
-            return;
-        }
-
-        Player firstPlayer = resolvePlayer(promptUserForName());
-        if (!isValidPlayer(firstPlayer)) {
-            return;
+            throw new EmptyDatabaseException("Players database is empty.");
         }
+        playersDatabase.print();
 
-        Player secondPlayer = resolvePlayer(promptUserForName());
-        if (!isValidPlayer(secondPlayer)) {
-            return;
+        Optional<Player> firstPlayer = playersDatabase.findPlayer(Integer.parseInt(promptUserForId()));
+        if (firstPlayer.isEmpty()) {
+            throw new InvalidPlayerIdException("Can`t find player in database according to given player ID.");
         }
-
-        match = new Match(firstPlayer, secondPlayer);
-        System.out.println("\n" + firstPlayer.getName() + " is playing against "
-                + secondPlayer.getName() + "\n");
-    }
-
-    private boolean isValidPlayer(Player player) {
-        if (player == null) {
-            System.out.println("Can`t find player in database. Please enter valid name.");
-            return false;
+        Optional<Player> secondPlayer = playersDatabase.findPlayer(Integer.parseInt(promptUserForId()));
+        if (secondPlayer.isEmpty()) {
+            throw new InvalidPlayerIdException("Can`t find player in database according to given player ID.");
         }
-        System.out.println("Selected " + player.getName() + " for a match.");
-        return true;
-    }
 
-    private Player resolvePlayer(String name) {
-        return playersDatabase.findPlayer(name);
+        match = new Match(firstPlayer.get(), secondPlayer.get());
+        System.out.println("\n" + firstPlayer.get().getName() + " is playing against "
+                + secondPlayer.get().getName() + "\n");
     }
 
     private void simulateMatch() {
-        selectPlayersForMatchSimulation();
-        if (match == null) {
+        try {
+            selectPlayersForMatchSimulation();
+            match.simulate();
+            print();
+            match = null;
+        } catch (RuntimeException e) {
```

Comment:
Specify both exceptions which you are actually expecting here instead of catching RuntimeException. If some random exception which extends RuntimeException would be thrown it would also be caught here and that's not a good idea

Reply:
fixed